components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/OutboundProvisioningManager.java

```diff
@@ -401,12 +401,29 @@ public void provision(ProvisioningEntity provisioningEntity, String serviceProvi
                                                             + idPName);
                 }
 
-                String outboundClaimDialect = connector.getClaimDialectUri();
+                String[] outboundClaimDialects = connector.getClaimDialectUris();
 
-                if (outboundClaimDialect == null
-                    && (provisioningIdp.getClaimConfig() == null || provisioningIdp
-                        .getClaimConfig().isLocalClaimDialect())) {
-                    outboundClaimDialect = IdentityProvisioningConstants.WSO2_CARBON_DIALECT;
+                if (outboundClaimDialects == null) {
+                    String outboundClaimDialect = connector.getClaimDialectUri();
+                    if (outboundClaimDialect == null) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("No outbound claim dialects available for the connector.");
+                        }
+                        if (provisioningIdp.getClaimConfig() == null || provisioningIdp
+                                .getClaimConfig().isLocalClaimDialect()) {
+                            outboundClaimDialect = IdentityProvisioningConstants.WSO2_CARBON_DIALECT;
+                        }
+                    } else {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Outbound claim dialect available for the connector: " + outboundClaimDialect);
```

Comment:
Can we add the connector name as well
'Outbound dialect: dialect + for the connector:  connector name'